tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -222,6 +265,106 @@ export class ScalarCardContainer implements CardRenderer, OnInit {
       distinctUntilChanged(areSeriesDataListEqual)
     );
 
+    function getSmoothedSeriesId(seriesId: string): string {
+      return JSON.stringify(['smoothed', seriesId]);
+    }
+
+    this.dataSeries$ = runIdAndPoints$.pipe(
+      combineLatestWith(this.store.select(getMetricsScalarSmoothing)),
+      switchMap(([runsData, smoothing]) => {
+        const dataSeriesList = runsData.map(({runId, points}) => {
+          return {id: runId, points};
+        });
+        if (smoothing === 0) {
+          return of(dataSeriesList);
+        }
+        return from(classicSmoothing(dataSeriesList, smoothing)).pipe(
+          map((smoothedDataSeriesList) => {
+            const dataSeriesMap = new Map(
+              dataSeriesList.map((datum) => [datum.id, datum])
+            );
+            for (const smoothedSeries of smoothedDataSeriesList) {
+              const {points: origPoints} = dataSeriesMap.get(
+                smoothedSeries.id
+              )!;
```

Comment:
Given that the smoothing transformer returns a list, can we simplify by relying on the assumption that `classicSmoothing` returns a list in the same order as the input list?

e.g. dropping `dataSeriesMap` and do something like

```ts
const smoothedList = dataSeriesList.map((dataSeries, index) => {
  return {
    id: getSmoothedSeriesId(dataSeries.id),
    points: smoothedResults[index].map(/* add extra fields from original dataSeries.points */),
  };
});
return [...dataSeriesList, ...smoothedList];
```
?